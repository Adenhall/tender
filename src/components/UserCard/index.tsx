import React from 'react';
import { makeStyles, IconButton } from '@material-ui/core';
import FavoriteIcon from '@material-ui/icons/Favorite';
import CloseIcon from '@material-ui/icons/Close';
// import ShareIcon from "@material-ui/icons/Share";
// import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
// import MoreVertIcon from "@material-ui/icons/MoreVert";

const useStyles = makeStyles(() => ({
  container: {
    width: '100%',
    height: '100%',
    padding: '1rem',
  },
  header: {
    display: 'flex',
    marginBottom: '.3rem',
    '& :last-child': {
      marginLeft: '.3rem',
    },
  },
  mediaContainer: {
    '& img': {
      width: '8rem',
    },
  },
  biography: {
    marginTop: '6rem',
    height: '300px',
    width: '80%',
    overflowY: 'auto',
    wordBreak: 'break-all',
  },
  actions: {
    display: 'flex',
    position: 'absolute',
    bottom: '0',
  },
}));

type UserCardProps = {
  id: string;
  name: string;
  age?: number;
  profilePic: string;
  bio?: string;
  handleLike: (id: string) => void;
  handlePass: (id: string, dir?: number) => void;
};

const UserCard: React.FC<UserCardProps> = ({
  name, age = 0, profilePic, bio, handleLike, handlePass, id,
}: UserCardProps) => {
  const classes = useStyles();
  return (
    <div className={classes.container}>
      <div className={classes.header}>
        <div>
          {name}
          ,
        </div>
        <div>{age}</div>
      </div>
      <div className={classes.mediaContainer}>
        <img alt={name} src={profilePic} />
      </div>
      <div className={classes.biography}>{bio || 'This one does not have any info or lame pick-up lines. Boringggg'}</div>
      <div className={classes.actions}>
        <IconButton onClick={() => handleLike(id)}>
          <FavoriteIcon />
        </IconButton>
        <IconButton onClick={() => handlePass(id, 1)}>
          <CloseIcon />
        </IconButton>
      </div>
    </div>
  );
};

export default UserCard;
