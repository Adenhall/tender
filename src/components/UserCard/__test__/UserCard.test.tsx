/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */

import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import UserCard from '../index';

describe('UserCard component', () => {
  let container: any = null;

  const userCardProps = {
    handleLike: () => '',
    handlePass: () => {},
    id: '',
    name: 'James',
    profilePic: '',
    bio: 'Hey, stranger',
  };

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
  it('renders without crashing', () => {
    act(() => {
      render(<UserCard {...userCardProps} />, container);
    });
    expect(container.textContent).toBe(`${userCardProps.name},${userCardProps.bio}`);
  });

  it('matches snapshot', () => {
    act(() => {
      render(<UserCard {...userCardProps} />, container);
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
