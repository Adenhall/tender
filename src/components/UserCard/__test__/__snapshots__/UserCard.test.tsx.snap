// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component matches snapshot 1`] = `
"<div class=\\"makeStyles-container-6\\">
  <div class=\\"makeStyles-header-7\\">
    <div>James,</div>
    <div></div>
  </div>
  <div class=\\"makeStyles-mediaContainer-8\\"><img alt=\\"James\\" src=\\"\\"></div>
  <div class=\\"makeStyles-biography-9\\">Hey, stranger</div>
  <div class=\\"makeStyles-actions-10\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
</div>"
`;
